integration/spark-common-test/src/test/scala/org/apache/carbondata/integration/spark/testsuite/timeseries/TestTimeseriesTableSelection.scala

```diff
@@ -99,124 +113,763 @@ class TestTimeseriesTableSelection extends QueryTest with BeforeAndAfterAll {
     sql(s"LOAD DATA LOCAL INPATH '$resourcesPath/timeseriestest.csv' into table mainTable")
   }
 
-  test("test PreAggregate table selection 1") {
-    val df = sql("select mytime from mainTable group by mytime")
+  test("test timeseries table selection 1") {
+    val df = sql("SELECT mytime FROM mainTable GROUP BY mytime")
     preAggTableValidator(df.queryExecution.analyzed, "maintable")
   }
 
-  test("test PreAggregate table selection 2") {
-    val df = sql("select timeseries(mytime,'hour') from mainTable group by timeseries(mytime,'hour')")
+  test("test timeseries table selection 2") {
+    val df = sql("SELECT TIMESERIES(mytime,'hour') FROM mainTable GROUP BY TIMESERIES(mytime,'hour')")
     preAggTableValidator(df.queryExecution.analyzed, "maintable_agg0_hour")
   }
 
-  test("test PreAggregate table selection 3") {
-    val df = sql("select timeseries(mytime,'milli') from mainTable group by timeseries(mytime,'milli')")
-    preAggTableValidator(df.queryExecution.analyzed, "maintable")
+  test("test timeseries table selection 3: No enum constant MILLI") {
+    val e = intercept[Exception] {
+      val df = sql(
+        """
+          | SELECT TIMESERIES(mytime,'milli')
+          | FROM mainTable
+          | GROUP BY TIMESERIES(mytime,'milli')
+        """.stripMargin)
+      preAggTableValidator(df.queryExecution.analyzed, "maintable")
+      df.show()
+    }
+    assert(e.getMessage.contains(
+      "No enum constant org.apache.carbondata.core.preagg.TimeSeriesFunctionEnum.MILLI"))
   }
 
-  test("test PreAggregate table selection 4") {
-    val df = sql("select timeseries(mytime,'year') from mainTable group by timeseries(mytime,'year')")
+  test("test timeseries table selection 4") {
+    val df = sql("SELECT TIMESERIES(mytime,'year') FROM mainTable GROUP BY TIMESERIES(mytime,'year')")
```

Comment:
why is this requried?

Reply:
The operation should be upper case and table name should be down case according to SQL syntax, Hive syntax also like this.